mungegithub/mungers/approvers/owners_test.go

```diff
@@ -669,3 +669,76 @@ func TestRemoveSubdirs(t *testing.T) {
 		}
 	}
 }
+
+func TestGetApprovalStatus(t *testing.T) {
+	tests := []struct {
+		testName       string
+		filenames      []string
+		approvers      []string
+		owners         map[string]sets.String
+		expectedStatus map[string]sets.String
+	}{
+		{
+			testName:       "Empty PR",
+			filenames:      []string{},
+			approvers:      []string{},
+			owners:         map[string]sets.String{},
+			expectedStatus: map[string]sets.String{},
+		},
+		{
+			testName:       "No approvers",
+			filenames:      []string{"a/a", "c"},
+			approvers:      []string{},
+			owners:         map[string]sets.String{"": sets.NewString("RootOwner")},
+			expectedStatus: map[string]sets.String{"": sets.String{}},
+		},
+		{
+			testName: "Approvers approves some",
+			filenames: []string{
+				"a/a",
+				"c/c",
+			},
+			approvers: []string{"CApprover"},
+			owners: map[string]sets.String{
+				"a": sets.NewString("AApprover"),
+				"c": sets.NewString("CApprover"),
+			},
+			expectedStatus: map[string]sets.String{
+				"a": sets.String{},
+				"c": sets.NewString("CApprover"),
+			},
+		},
+		{
+			testName: "Multiple approvers",
+			filenames: []string{
+				"a/a",
+				"c/c",
+			},
+			approvers: []string{"RootApprover", "CApprover"},
+			owners: map[string]sets.String{
+				"":  sets.NewString("RootApprover"),
+				"a": sets.NewString("AApprover"),
+				"c": sets.NewString("CApprover"),
+			},
+			expectedStatus: map[string]sets.String{
+				"a": sets.NewString("RootApprover"),
+				"c": sets.NewString("RootApprover", "CApprover"),
+			},
+		},
+		{
+			testName:       "Case-insensitive approvers",
```

Comment:
which one is supposed to be reflected in the owners files?  Theoretically it should work bidirectionally right?  (Weird caps in owners files or weird capitalization in the comment list).  Can you add a test for the reverse as well? And which should be displayed "RootApprover" or "rOOtaPProver".